File: java/src/Ex4.java
public class Ex4 {

    static int scan_int(String error) {

        int value;

        while (true) {
            try {
                value = EntradaTeclado.leInt();
                break;
            } catch (Exception e) {
                System.out.println(error);
            }
        }

        return value;
    }

    public static void main(String[] args){

        // Prints right skewed pyramid to console given total height from user
        System.out.println("Insira o tamanho da piramide: ");

        int size = scan_int("Valor invalido inserido, tente novamente:");

        for (int i = size; i > 0; i--) {
            System.out.print(" ".repeat(size - i));
            System.out.println("*".repeat(i));
        }
    }
}

File: python/oop2022/exs1/ex4.py
def read_int(name):
    while True:
        try:
            value = int(input("Insira o valor {}:\n".format(name)))
            return value
        except ValueError:
            print("Valor invalido inserido, tente novamente:")


def main():
    size = read_int("do tamanho da piramide")

    for x in range(size, 0, -1):
        print(" " * (size - x) + "*" * x)


if __name__ == "__main__":
    main()
